delete-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusDeleteHandlerTest.java

```diff
@@ -0,0 +1,133 @@
+import com.amazonaws.services.lambda.runtime.events.models.s3.S3EventNotification.S3EventNotificationRecord;
+import com.amazonaws.services.lambda.runtime.events.models.s3.S3EventNotification.S3ObjectEntity;
+import com.amazonaws.services.lambda.runtime.events.models.s3.S3EventNotification.UserIdentityEntity;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.List;
+
+import no.unit.nva.commons.json.JsonUtils;
+import no.unit.nva.s3.S3Driver;
+import no.unit.nva.stubs.FakeEventBridgeClient;
+import no.unit.nva.stubs.FakeS3Client;
+import nva.commons.core.ioutils.IoUtils;
+import nva.commons.core.paths.UnixPath;
+import nva.commons.core.paths.UriWrapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class ScopusDeleteHandlerTest {
+
+    public static final Context CONTEXT = mock(Context.class);
+    public static final RequestParametersEntity EMPTY_REQUEST_PARAMETERS = null;
+    public static final ResponseElementsEntity EMPTY_RESPONSE_ELEMENTS = null;
+    public static final UserIdentityEntity EMPTY_USER_IDENTITY = null;
+    public static final long SOME_FILE_SIZE = 100L;
+
+    public static final String DELETE_MULTIPLE_IDENTIFIERS_FILE = "deleteMultipleIdentifiers.txt";
+    public static final String DELETE_SINGLE_IDENTIFIER_FILE = "deleteSingleIdentifier.txt";
+    public static final String HARDCODED_SCOPUS_IDENTIFIER_IN_DELETE_SINGLE_FILE = "0000687314";
+    public static final List<String> HARDCODED_SCOPUS_IDENTIFIERS_IN_DELETE_MULTIPLE_FILE =
+            Arrays.asList("0000687314", "0017752991", "33947568076", "85084744636");
+
+    private FakeS3Client s3Client;
+    private S3Driver s3Driver;
+    private ScopusDeleteHandler scopusDeleteHandler;
+    private FakeEventBridgeClient fakeEventBridgeClient;
+
+
+    @BeforeEach
+    public void init() {
+        s3Client = new FakeS3Client();
+        s3Driver = new S3Driver(s3Client, "ignoredValue");
+        fakeEventBridgeClient = new FakeEventBridgeClient();
+        scopusDeleteHandler = new ScopusDeleteHandler(s3Client, fakeEventBridgeClient);
+    }
+
+    @Test
+    void shouldEmitEventWithSingleScopusIdentifierWhenInputFileContainsSingleScopusIdentifier() throws IOException {
+        var scopusFile = IoUtils.stringFromResources(Path.of(DELETE_SINGLE_IDENTIFIER_FILE));
+        var uri = s3Driver.insertFile(UnixPath.of(randomString()), scopusFile);
+        var s3Event = createS3Event(uri);
+        scopusDeleteHandler.handleRequest(s3Event, CONTEXT);
+        var requestEntries = fakeEventBridgeClient.getRequestEntries();
+        assertThat(requestEntries.size(), is(equalTo(1)));
+        var eventBodyJson = requestEntries.get(0).detail();
+        var actualRequestEntryBody =
+                JsonUtils.dtoObjectMapper.readValue(eventBodyJson, ScopusDeleteEventBody.class);
```

Comment:
extract to method

Reply:
I refactored a little, but I'm not sure if it is what you had in mind